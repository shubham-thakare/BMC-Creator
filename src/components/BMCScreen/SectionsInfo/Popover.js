import React from "react";
import PropTypes from "prop-types";
import {
  Popover as BPPopover,
  PopoverInteractionKind
} from "@blueprintjs/core";

export const Popover = ({ children, content }) => {
  return (
    <BPPopover
      interactionKind={PopoverInteractionKind.HOVER}
      content={content}
      transitionDuration={100}
      lazy={true}
    >
      {children}
    </BPPopover>
  );
};

Popover.propTypes = {
  content: PropTypes.node
};
