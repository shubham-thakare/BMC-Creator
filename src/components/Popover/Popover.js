import React from 'react';
import PropTypes from 'prop-types';
import {
  Popover as BPPopover,
  PopoverInteractionKind,
} from '@blueprintjs/core';

const Popover = ({ children, content, ...props }) => {
  return (
    <BPPopover
      interactionKind={PopoverInteractionKind.CLICK}
      content={content}
      transitionDuration={100}
      lazy={true}
      {...props}
    >
      {children}
    </BPPopover>
  );
};

Popover.propTypes = {
  content: PropTypes.node,
};

export default Popover;
