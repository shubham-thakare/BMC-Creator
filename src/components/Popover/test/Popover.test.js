import React from 'react';
import { shallow } from 'enzyme';
import Popover from '..';

let wrapper;

const props = {
  content: <>Content</>,
  children: <>Children</>,
};

describe('Popover Component', () => {
  beforeAll(() => {
    wrapper = shallow(<Popover {...props} />);
  });

  it('To match snapshot of Popover', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
