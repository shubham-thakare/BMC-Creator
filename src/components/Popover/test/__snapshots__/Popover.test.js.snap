// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover Component To match snapshot of Popover 1`] = `
<Blueprint3.Popover
  boundary="scrollParent"
  captureDismiss={false}
  content={
    <React.Fragment>
      Content
    </React.Fragment>
  }
  defaultIsOpen={false}
  disabled={false}
  fill={false}
  hasBackdrop={false}
  hoverCloseDelay={300}
  hoverOpenDelay={150}
  inheritDarkTheme={true}
  interactionKind="hover"
  lazy={true}
  minimal={false}
  modifiers={Object {}}
  openOnTargetFocus={true}
  position="auto"
  targetTagName="span"
  transitionDuration={100}
  usePortal={true}
  wrapperTagName="span"
>
  Children
</Blueprint3.Popover>
`;
